homeassistant/components/sensor/bbox.py

```diff
@@ -0,0 +1,144 @@
+"""
+Support for Bbox Bouygues Modem Routeur.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.bbox/
+"""
+import logging
+from datetime import timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (BANDWIDTH_MEGABITS_SECONDS, CONF_NAME,
+                                 CONF_MONITORED_VARIABLES, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+REQUIREMENTS = ['pybbox==0.0.5-alpha']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Powered by Bouygues Telecom"
+DEFAULT_NAME = 'Bbox'
+
+# Sensor types are defined like so:
+# Name, unit, icon
+SENSOR_TYPES = {
+    'down_max_bandwidth': ['Maximum Download Bandwidth',
+                           BANDWIDTH_MEGABITS_SECONDS, 'mdi:download'],
+    'up_max_bandwidth': ['Maximum Upload Bandwidth',
+                         BANDWIDTH_MEGABITS_SECONDS, 'mdi:upload'],
+    'current_down_bandwidth': ['Currently Used Download Bandwidth',
+                               BANDWIDTH_MEGABITS_SECONDS, 'mdi:download'],
+    'current_up_bandwidth': ['Currently Used Upload Bandwidth',
+                             BANDWIDTH_MEGABITS_SECONDS, 'mdi:upload'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_VARIABLES):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# pylint: disable=too-many-arguments
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Bbox sensor."""
+    # Create a data fetcher to support all of the configured sensors. Then make
+    # the first call to init the data.
+    try:
+        bbox_data = BboxData()
```

Comment:
No because the bbox is provided to be plug and play so IP the box give to itself is always 192.168.1.254 .  And credentials are only needed for critical operation (reboot of the router for example). Restriction is you can only use this API on your local network